node_modules/oae-authentication/lib/strategies/local/init.js

```diff
@@ -79,7 +80,14 @@ module.exports = function(config) {
                     var strategyId = AuthenticationUtil.getStrategyId(tenant, AuthenticationConstants.providers.LOCAL);
                     var authObj = {'user': user, 'strategyId': strategyId};
                     AuthenticationUtil.logAuthenticationSuccess(req, authObj, AuthenticationConstants.providers.LOCAL);
-                    return done(null, authObj);
+
+                    // Save the date of the last connection into Cassandra 
```

Comment:
Why are we doing this only for local authentication? If we are to record this information, then it shouldn't matter which authentication was used.

Also, I'd prefer if we didn't do this like this, explicitly on the method - and if we are to add this to other strategies, it would then require some form of repetition. Instead, let's use events for this. Emit an event and handle it somewhere else (another file) in order to record it.

Reply:
Yeah ... it was stupid ... I changed that, tell me if it's ok : 

in oae-authentication/lib/api.js I caught the event "USER_LOGGED_IN" and then I save the log date into Cassandra. 

Is it a good place or should I change it ?